### PR TITLE
Bump 1.23.0

### PR DESCRIPTION
**Pull request by rhatdan**
#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by nalind** [comment]
Are we planning to cut a v1.23.1 from the main branch later?  If not, this should leave it marked as v1.24.0-dev.

File: release.sh [hunk 1]
```diff
@@ -83,7 +83,7 @@ epoch_commit()
 }
 
 git fetch origin &&
-git checkout -b "bump-${VERSION}" origin/master &&
```
Comment (TomSweeneyRedHat) [inline release.sh:86]: nice catch, wonder how this slipped through?

**Comment by TomSweeneyRedHat** [comment]
Need to update CHANGELOG.md too

**Comment by TomSweeneyRedHat** [comment]
And you need to add "NO TESTS NEEDED" to the description, my nemesis.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
Fedora 33 w/overlay isn't happy, I don't think it's a flake.

```
[+1146s] # time="2021-09-13T16:54:07-05:00" level=error msg="A layer is mounted: layer is in use by a container"
[+1146s] # $ /var/tmp/go/src/github.com/containers/buildah/tests/./../bin/buildah from --signature-policy /var/tmp/go/src/github.com/containers/buildah/tests/./policy.json --quiet --userns-uid-map-user=foo --userns-gid-map-group=fedora alpine
[+1146s] # alpine-working-container-5
[+1146s] # $ /var/tmp/go/src/github.com/containers/buildah/tests/./../bin/buildah run --cni-config-dir=/var/tmp/buildah_tests.kgodn2/no-cni-configs alpine-working-container-5 readlink /proc/self/ns/user
[+1146s] # user:[4026532328]
[+1146s] # $ /var/tmp/go/src/github.com/containers/buildah/tests/./../bin/buildah run --cni-config-dir=/var/tmp/buildah_tests.kgodn2/no-cni-configs alpine-working-container-5 cat /proc/self/uid_map
[+1146s] #          0     165536      65536
[+1146s] # $ /var/tmp/go/src/github.com/containers/buildah/tests/./../bin/buildah run --cni-config-dir=/var/tmp/buildah_tests.kgodn2/no-cni-configs alpine-working-container-5 cat /proc/self/gid_map
[+1146s] #          0     231072      65536
[+1146s] #      65536     296608      65536
[+1146s] # #/vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
[+1146s] # #|     FAIL: gid_map
[+1146s] # #| expected: '0 231072
[+1146s] # 296608 65536'
[+1146s] # #|   actual: '0 231072 65536'
[+1146s] # #|         > '65536 296608 65536'
[+1146s] # #\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[+1146s] # /var/tmp/go/src/github.com/containers/buildah/tests
```

**Pull request merged**